### DISCUSSION TITLE
Uncaught TypeError: Cannot delete property 'Prism' of #<Window>

### DISCUSSION THREAD

**Discussion by jcubic** [facebook/docusaurus#9870] [Q&A] [answered]
Got this error from the minified code from the main Docusourus bundle.

```javascript
    3310: (e,t,n)=>{
        "use strict";
        n.r(t);
        var r = n(2573)
          , a = n(6809);
        !function(e) {
            const {themeConfig: {prism: t}} = a.default
              , {additionalLanguages: r} = t;
            globalThis.Prism = e,
            r.forEach((e=>{
                "php" === e && n(6854),
                n(4280)(`./prism-${e}`)
            }
            )),
            delete globalThis.Prism
        }(r.p1)
    }
```

Note that I use PrismJS on my own and I need to be global, I'm not sure why there is an error. I have this script tag loaded with the Helmet:

Here is DevBox: https://codesandbox.io/p/devbox/lucid-lake-gjnj4q?file=%2Fsrc%2Fpages%2Findex.tsx%3A37%2C47

The error is in the browser console, it's not visible on the output page.

**EDIT**: If I remove this code before build:

```bash
sed -i '/delete globalThis.Prism/d' node_modules/@docusaurus/theme-classic/lib/theme/prism-include-languages.js
```

Got this errors in console:

> Docusaurus React Root onRecoverableError: Error: Minified React error #418; visit https://reactjs.org/docs/error-decoder.html?invariant=418 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

**Comment by slorber** [comment]
> Note that I use PrismJS on my own and I need to be global

I don't know why you need that, but I wouldn't recommend this and I can't guarantee this will work well. 

If you want to keep loading 2 competing versions of Prism on the same site, you have to figure out how yourself unfortunately

**Reply by jcubic** [reply]
I would love to use Prism that came with Docusourus but you delete it after use.

The problem is that Docusousurs make the object global even if it already exits and delete someone it doesn't own.

**Comment by Josh-Cena** [comment, marked as answer]
Hello, you can totally achieve this via [swizzling](https://docusaurus.io/docs/swizzling). You would eject `prism-include-languages` and delete the line where it says `delete globalThis.Prism`. Alternatively, the `Prism` global object is imported from `import {Prism} from 'prism-react-renderer';` (see here: https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/prism-include-languages.ts), so in your own script, you could also import that instance and do `globalThis.Prism = Prism;` and it will work :) Alternatively, if you only depend on `Prism` being global _for some time_ but not always (for example, you are loading another component or plugin), you can put your logic directly inside the ejected `prism-include-languages` code without changing anything else!

**Reply by jcubic** [reply]
Thanks, this seems to work. I still have issues with using Prism, but this particular issue is solved.
